lib/game.rb

```diff
@@ -73,12 +76,95 @@ def change_current_player
   def is_valid?(input)
     is_valid_column?(input) && !is_column_full?(input)
   end
-
+  
   def is_valid_column?(input)
     %w[A B C D E F G].include?(input)
   end
-
+  
   def is_column_full?(input)
     @board.visual[input].none?(".")
   end
+
+  def win?
+    if sw_to_ne_diagonal_win? || se_to_nw_diagonal_win? || horizontal_win? || vertical_win?
+      'True'
+    else
+      'False'
+    end
+  end
+
+  def vertical_win?
```

Comment:
Great job transforming what we figured out with our mentor into code that worked, even accounting for the first turn issues we ran into.